### PR TITLE
Allow defining the entire strategy.matrix value

### PR DESCRIPTION
**Pull request by elesel**
* Add `addInclude` input
* Update documentation

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -3,12 +3,16 @@ author: "JoshuaTheMiller"
...
+    description: 'The file that contains the JSON configuration of the matrix elements. Must come after a Checkout.'
     required: true
     default: '.github/workflows/matrix_includes.json'
   filter: 
-    description: 'The JMESPATH JSON query to apply to the input file.'
+    description: 'The JMESPath JSON query to apply to the input file.'
     required: true
+  addInclude:
+    description: 'Wrap the output from the JSON query in an "{include: ...}" object.'
+    required: true
```
Comment (JoshuaTheMiller) [inline action.yml]:
```suggestion
    required: false
```

Setting `required` to `true` would be a breaking change (in terms of expectations) for existing users. Adding this input with a default value of `true` allows for folks to override the value while not changing the contract altogether 🙂

File: index.js [hunk 2]
```diff
@@ -3,18 +3,18 @@ const jmes = require('jmespath')
...
 
     const inputJson = JSON.parse(inputFile);
 
-    const filteredIncludes = jmes.search(inputJson, filterString);
-    const filteredIncludesAsString = JSON.stringify(filteredIncludes);
+    const filteredJson = jmes.search(inputJson, filterString);
+    const filteredJsonAsString = JSON.stringify(filteredJson);
 
-    const outputString = `{"include":${filteredIncludesAsString}}`
+    const outputString = addInclude ? `{"include":${filteredJsonAsString}}` : filteredJsonAsString;
```
Comment (JoshuaTheMiller) [inline index.js:17]: Good call on the name changes here 👍

File: index.js [hunk 3]
```diff
@@ -3,18 +3,18 @@ const jmes = require('jmespath')
 const fs = require('fs')
 
 try {
-    // `who-to-greet` input defined in action metadata file
```
Comment (JoshuaTheMiller) [inline index.js:6]: Thanks for the cleanup 🙏

File: README.md [hunk 4]
```diff
@@ -103,7 +105,9 @@ jobs:
 
 ### Necessary JSON File
 
-This Action requires a JSON file that contains the information you would normally include in a `strategy.matrix.includes` block (example is [included below](#sample-json-file)). By default, the action will look for file in your `.github/workflows/` folder named `matrix_includes.json`. If you want to name the file differently, or place it in a different folder, set the path via the `inputFile` input.
+This Action requires a JSON file that contains the information you would normally include in a `strategy.matrix.include` block (example is [included below](#sample-json-file)). By default, the action will look for a file in your `.github/workflows/` folder named `matrix_includes.json`. If you want to name the file differently, or place it in a different folder, set the path via the `inputFile` input.
+
+If you need to define the entire `strategy.matrix` you'll need to set `addInclude` to `false` and then ensure your input file and JSON query provide the results your workflow needs.
```
Comment (JoshuaTheMiller) [inline README.md]:
```suggestion
If you need to define the entire `strategy.matrix` value, you will need to set `addInclude` to `false`. Please note that you must then ensure your **input file** and **JSON query** provide the results your workflow needs.
```

Some changes to match style

**Review by JoshuaTheMiller — CHANGES_REQUESTED** [review]
Thanks for the contribution! 

Please see my suggested changes and commit/accept them unless you disagree with the changes. In that case, please add a reply.

If I don't hear back in a week (or longer), I will commit the changes and merge this PR as I find it to be beneficial 🙂 (I will consider adding this policy to the contribution guidelines depending on how well the policy works).

**Comment by elesel** [comment]
> Thanks for the contribution!
> 
> Please see my suggested changes and commit/accept them unless you disagree with the changes. In that case, please add a reply.
> 
> If I don't hear back in a week (or longer), I will commit the changes and merge this PR as I find it to be beneficial 🙂 (I will consider adding this policy to the contribution guidelines depending on how well the policy works).

Thanks for considering these changes! Might I recommend they be tagged as version `0.0.2` so folks can target specific versions within their workflows?

**Comment by JoshuaTheMiller** [comment]
> > Thanks for the contribution!
> > Please see my suggested changes and commit/accept them unless you disagree with the changes. In that case, please add a reply.
> > If I don't hear back in a week (or longer), I will commit the changes and merge this PR as I find it to be beneficial 🙂 (I will consider adding this policy to the contribution guidelines depending on how well the policy works).
> 
> Thanks for considering these changes! Might I recommend they be tagged as version `0.0.2` so folks can target specific versions within their workflows?

I will definitely increment the version tag once this is merged! 

(I'm expecting that I will tag this as `0.1.0` as it [adds new functionality](https://semver.org/))

**Pull request merged**